### ISSUE TITLE
Make Structure.members a dict instead of a list

### ISSUE THREAD

**Issue by joshwatson** [Vector35/binaryninja-api#1332]
`Structure.members` should return a `dict` keyed on the name of the `StructureMember` objects, instead of a list. This would make it much easier to find a particular member. Instead of something like:

```python
a = next(m for m in struct.members if m.name == 'a')
```

One could just do:
```python
a = struct.members['a']
```

**Comment by plafosse** [comment]
I feel like just implementing a `__getitem__` on the Structure class is a much better option, and it doesn't break API compatibility.

**Comment by plafosse** [comment]
... and we should add a `BNGetStructureMemberByName` core API otherwise we have to pull over the whole list each time.

**Comment by joshwatson** [comment]
Actually yeah that's a way better solution!

**Comment by plafosse** [comment]
Fixed in > v1.1.1615
